### PR TITLE
Add error handling to PATCH /api/reviews/:review_id

### PR DESCRIPTION
**Pull request by thatnick**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -70,3 +70,55 @@ describe("GET /api/reviews/:review_id", () => {
     });
   });
 });
+
+describe("PATCH /api/reviews/:review_id", () => {
+  test("status: 202, in/decrements votes column for the given review_id by the value of inc_votes, responds with updated review", async () => {
+    const { body } = await request(app)
+      .patch("/api/reviews/2")
+      .send({ inc_votes: -1 })
+      .expect(202);
```
Comment (douglashellowell) [inline __tests__/app.test.js:79]: `202 Accepted` doesn't mean that the request has been done - just that the server has accepted it!

you might want to go for the tried and true `200` here

[read more](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -70,3 +70,55 @@ describe("GET /api/reviews/:review_id", () => {
...
+      created_at: "2021-01-18T10:01:41.251Z",
+      votes: 4,
+    });
+  });
+
+  test("status: 400, responds with error if a review with the given id does not exist", async () => {
+    const { body } = await request(app)
+      .patch("/api/reviews/14")
+      .send({ inc_votes: 1 })
+      .expect(400);
```
Comment (douglashellowell) [inline __tests__/app.test.js:99]: (spoilers for tomorrows chat)

i'd go for a `404` here!
You're asking to access a resource and it doesn't exist - the request itself was totally valid however

File: __tests__/app.test.js (hunk 1 above)
Comment (thatnick) [reply __tests__/app.test.js]: Changed it to 200, thanks. So we'd use 202 if the server had queued the patch to do later, or something along those lines?

File: __tests__/app.test.js (hunk 2 above)
Comment (thatnick) [reply __tests__/app.test.js]: Got it! Changed to a 404.

File: model/categories.model.js [hunk 3]
```diff
@@ -35,3 +35,32 @@ exports.selectReviewById = async (review_id) => {
     return Promise.reject();
   }
 };
+
+exports.updateReviewById = async (review_id, inc_votes) => {
+  try {
+    const {
```
Comment (douglashellowell) [inline model/categories.model.js:41]: this is some super fancy code!

nested destructuring, with some array destructuring, ternary opertator all wrapped up in a `try`/`catch` `async`/`await` bow? 

*chefs kiss* 🤌

There is something to be said about readability here however - sometimes a good old `if` statement can make you and your teammates lives a little bit easier

Comment (thatnick) [reply model/categories.model.js]: Thanks! Good point about readability, I split up some of the nested destructuring to help with that.

**Pull request merged**